espnet2/bin/s2t_inference.py

```diff
@@ -0,0 +1,1040 @@
+        quantize_dtype = getattr(torch, quantize_dtype)
+
+        # 1. Build S2T model
+        s2t_model, s2t_train_args = S2TTask.build_model_from_file(
+            s2t_train_config, s2t_model_file, device
+        )
+        s2t_model.to(dtype=getattr(torch, dtype)).eval()
+
+        if quantize_s2t_model:
+            logging.info("Use quantized s2t model for decoding.")
+
+            s2t_model = torch.quantization.quantize_dynamic(
+                s2t_model, qconfig_spec=quantize_modules, dtype=quantize_dtype
+            )
+
+        decoder = s2t_model.decoder
+        ctc = CTCPrefixScorer(ctc=s2t_model.ctc, eos=s2t_model.eos)
+        token_list = s2t_model.token_list
+        scorers = dict(
+            decoder=decoder,
+            ctc=ctc,
+            length_bonus=LengthBonus(len(token_list)),
+        )
+
+        # 2. Build language model
+        if lm_train_config is not None:
+            lm, lm_train_args = LMTask.build_model_from_file(
+                lm_train_config, lm_file, device
+            )
+
+            if quantize_lm:
+                logging.info("Use quantized lm for decoding.")
+
+                lm = torch.quantization.quantize_dynamic(
+                    lm, qconfig_spec=quantize_modules, dtype=quantize_dtype
+                )
+
+            scorers["lm"] = lm.lm
+
+        # 3. Build ngram model
+        if ngram_file is not None:
+            if ngram_scorer == "full":
+                from espnet.nets.scorers.ngram import NgramFullScorer
+
+                ngram = NgramFullScorer(ngram_file, token_list)
+            else:
+                from espnet.nets.scorers.ngram import NgramPartScorer
+
+                ngram = NgramPartScorer(ngram_file, token_list)
+            scorers["ngram"] = ngram
+
+        # 4. Build BeamSearch object
+        if s2t_model.use_transducer_decoder:
+            # In multi-blank RNNT, we assume all big blanks are
+            # just before the standard blank in token_list
+            multi_blank_durations = getattr(
+                s2t_model, "transducer_multi_blank_durations", []
+            )[::-1] + [1]
+            multi_blank_indices = [
+                s2t_model.blank_id - i + 1
+                for i in range(len(multi_blank_durations), 0, -1)
+            ]
+
+            if transducer_conf is None:
+                transducer_conf = {}
+
+            beam_search_transducer = BeamSearchTransducer(
+                decoder=s2t_model.decoder,
+                joint_network=s2t_model.joint_network,
+                beam_size=beam_size,
+                lm=scorers["lm"] if "lm" in scorers else None,
+                lm_weight=lm_weight,
+                multi_blank_durations=multi_blank_durations,
+                multi_blank_indices=multi_blank_indices,
+                token_list=token_list,
+                **transducer_conf,
+            )
+            beam_search = None
+            hugging_face_model = None
+            hugging_face_linear_in = None
```

Comment:
Please remove the unnecessary functions for this PR. non only for the case that we probably won't test those options in a single PR, but it makes the PR a bit massive. Could you please remove those?

Reply:
Thanks. I simplified s2t inference a bit.